addons/html_editor/static/src/main/media/image_transform_button.js

```diff
@@ -3,53 +3,41 @@ import { toolbarButtonProps } from "@html_editor/main/toolbar/toolbar";
 import { registry } from "@web/core/registry";
 import { ImageTransformation } from "./image_transformation";
 
-export class ImageTransformButton extends Component {
-    static template = "html_editor.ImageTransformButton";
-    static props = {
-        id: String,
-        icon: String,
-        title: String,
-        getTargetedImage: Function,
-        resetImageTransformation: Function,
-        addStep: Function,
-        document: { validate: (p) => p.nodeType === Node.DOCUMENT_NODE },
-        editable: { validate: (p) => p.nodeType === Node.ELEMENT_NODE },
-        ...toolbarButtonProps,
-        activeTitle: String,
-    };
+export function useTransformOperations({
+    document,
+    isImageTransformationOpen,
```

Comment:
Directly in the hook as it is something common between components that will use the hook?